### PR TITLE
Log trauma gain and loss

### PR DESCRIPTION
**Pull request by Time-Green**
Noticed it in https://github.com/tgstation/tgstation/issues/83493 

Logging good, lets us figure out bugs and stuff

### CODE REVIEW

File: code/datums/brain_damage/brain_trauma.dm [hunk 1]
```diff
@@ -22,6 +22,7 @@
 	// Handles our references with our brain
 	brain?.remove_trauma_from_traumas(src)
 	if(owner)
+		log_game("[key_name(owner)] has lost the following brain trauma: [type]")
```
Comment (san7890) [inline code/datums/brain_damage/brain_trauma.dm:25]:
```suggestion
		log_game("[key_name_and_tag(owner)] has lost the following brain trauma: [type]")
```

File: code/modules/mob/living/brain/brain_item.dm [hunk 2]
```diff
@@ -508,6 +508,7 @@
 		actual_trauma.owner = owner
 		SEND_SIGNAL(owner, COMSIG_CARBON_GAIN_TRAUMA, trauma)
 		actual_trauma.on_gain()
+		log_game("[key_name(owner)] has gained the following brain trauma: [trauma.type]")
```
Comment (san7890) [inline code/modules/mob/living/brain/brain_item.dm:511]:
```suggestion
		log_game("[key_name_and_tag(owner)] has gained the following brain trauma: [trauma.type]")
```

**Review by san7890 — CHANGES_REQUESTED** [review]
probably good to have the tag here as well as a good aide in case some REALLY weird shit happens

**Comment by Loafin34** [comment]
thanks

**Pull request merged**